source/extensions/access_loggers/http_grpc/grpc_access_log_impl.cc

```diff
@@ -61,17 +81,21 @@ void GrpcAccessLoggerImpl::log(envoy::data::accesslog::v2::HTTPAccessLogEntry&&
                            "envoy.service.accesslog.v2.AccessLogService.StreamAccessLogs"),
                        *stream_);
 
-    auto* identifier = message.mutable_identifier();
+    auto* identifier = message_.mutable_identifier();
     *identifier->mutable_node() = local_info_.node();
     identifier->set_log_name(log_name_);
   }
 
   if (stream_->stream_ != nullptr) {
-    stream_->stream_->sendMessage(message, false);
+    stream_->stream_->sendMessage(message_, false);
   } else {
     // Clear out the stream data due to stream creation failure.
     stream_.reset();
   }
+
+  // Clear the message regardless of the success.
+  approximate_message_size_bytes_ = 0;
```

Comment:
qq: could we potentially use the proto built-in cached size to avoid keeping track of it ourselves?

Reply:
I don't see how. `ByteSize()` (or `ByteSizeLong()`) method does not use cached sizes for children (presumably because `CachedSize` is not invalidated on message modification).
Which means that the only way to use built-in cache is to manually iterate over all log entries and sum result of `GetCachedSize()` calls, which would be less efficient and less obvious in terms of implementation.

Reply:
OK SGTM. Thanks for the explanation.